components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/services/ManagementProviderImpl.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.governance.taxonomy.services;
+
+import org.apache.axiom.om.OMElement;
+import org.wso2.carbon.governance.taxonomy.beans.TaxonomyBean;
+import org.wso2.carbon.registry.common.services.RegistryAbstractAdmin;
+import org.wso2.carbon.registry.core.*;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+import javax.xml.namespace.QName;
+import static org.wso2.carbon.governance.taxonomy.util.CommonUtils.buildOMElement;
+import static org.wso2.carbon.governance.taxonomy.util.TaxonomyConstants.TAXONOMY_CONFIGURATION_PATH;
+import static org.wso2.carbon.governance.taxonomy.util.TaxonomyConstants.TAXONOMY_MEDIA_TYPE;
+
+class ManagementProviderImpl extends RegistryAbstractAdmin implements IManagementProvider {
+
+    /**
+     * This method will add user defined taxonomy content into the registry
+     *
+     * @param storageProvider storage provider instance
+     * @param taxonomyBean    taxonomy bean instance which contains taxonomy meta data
+     * @return
+     * @throws RegistryException
+     */
+    @Override
+    public boolean addTaxonomy(IStorageProvider storageProvider, TaxonomyBean taxonomyBean)
+            throws RegistryException {
+        Registry registry = getGovernanceUserRegistry();
+        Resource resource;
+        String name;
+        OMElement element = null;
+        element = buildOMElement(taxonomyBean.getPayload());
+        name = element.getAttributeValue(new QName("name"));
+
+        if (!getGovernanceUserRegistry().resourceExists(TAXONOMY_CONFIGURATION_PATH + name)) {
```

Comment:
Can't we use the registry variable here , rather than calling getGovernanceUserRegistry() method again.